src/pudl/extract/excel.py

```diff
@@ -193,31 +195,34 @@ def extract(self, years):
                 logger.debug(f'Skipping blacklisted page {page}.')
                 continue
             df = pd.DataFrame()
-            for yr in years:
+            for partition in pudl.helpers.iterate_multivalue_dict(**partitions):
                 # we are going to skip
-                if self.excel_filename(yr, page) == '-1':
+                if self.excel_filename(page, **partition) == '-1':
```

Comment:
Now that I think of it, string valued `-1` is a very non-standard way of encoding missing value. We should just return None here.

Reply:
this is a very good point. i'm going to add an issue for that.